protocol/headers.go

```diff
@@ -42,114 +61,151 @@ type Headers struct {
 
 // CorrelationID returns the 'correlation-id' header value or empty string if not set.
 func (h *Headers) CorrelationID() string {
-	if h.Values[HeaderCorrelationID] == nil {
-		return ""
+	if value, ok := h.Values[HeaderCorrelationID]; ok && value != nil {
+		return value.(string)
 	}
-	return h.Values[HeaderCorrelationID].(string)
+	return ""
 }
 
-// Timeout returns the 'timeout' header value or empty string if not set.
-func (h *Headers) Timeout() string {
-	if h.Values[HeaderTimeout] == nil {
-		return ""
+// Timeout returns the 'timeout' header value or duration of 60 seconds if not set.
+func (h *Headers) Timeout() time.Duration {
+	if value, ok := h.Values[HeaderTimeout]; ok {
+		if duration, err := parseTimeout(value.(string)); err == nil {
+			return duration
+		}
+	}
+	return 60 * time.Second
+}
+
+func parseTimeout(timeout string) (time.Duration, error) {
+	l := len(timeout)
+	if l > 0 {
+		t := time.Duration(-1)
+		switch timeout[l-1] {
+		case 'm':
+			if i, err := strconv.Atoi(timeout[:l-1]); err == nil {
+				t = time.Duration(i) * time.Minute
+			}
+		case 's':
+			if timeout[l-2] == 'm' {
+				if i, err := strconv.Atoi(timeout[:l-2]); err == nil {
+					t = time.Duration(i) * time.Millisecond
+				}
+			} else {
+				if i, err := strconv.Atoi(timeout[:l-1]); err == nil {
+					t = time.Duration(i) * time.Second
+				}
+			}
+		default:
+			if i, err := strconv.Atoi(timeout); err == nil {
+				t = time.Duration(i) * time.Second
+			}
+		}
+		if inTimeoutRange(t) {
+			return t, nil
+		}
 	}
-	return h.Values[HeaderTimeout].(string)
+	return -1, fmt.Errorf("invalid timeout '%s'", timeout)
+}
+
+func inTimeoutRange(timeout time.Duration) bool {
+	return timeout >= 0 && timeout < time.Hour
 }
```

Comment:
If we are about to perform header values validation here - are we about to introduce them for all known headers limitations and expected values - e.g. requested-acks header value validation, etc.? Otherwise, it is not really consistent.